src/utils/device/dehydration.ts

```diff
@@ -45,8 +23,8 @@ export async function initialiseDehydrationIfEnabled(
     opts: StartDehydrationOpts = {},
 ): Promise<void> {
     const crypto = client.getCrypto();
-    if (await deviceDehydrationEnabled(client, crypto)) {
-        logger.log("Device dehydration enabled");
+    if (crypto && (await crypto.isDehydrationSupported())) {
```

Comment:
```suggestion
    if (await crypto?.isDehydrationSupported()) {
```

Reply:
we can't do both this *and* remove the `!` below: 

```
src/utils/device/dehydration.ts:28:15 - error TS18048: 'crypto' is possibly 'undefined'.

28         await crypto.startDehydration(opts);
                 ~~~~~~

```

Care to pick one?

Reply:
> we can't do both this and remove the ! below:

That's surprising, given `await undefined` wouldn't go into the if. I guess TS inference isn't perfect. I don't mind, your choice, neither is also fine.